src/batch-submitter/batch-submitter.ts

```diff
@@ -142,47 +153,48 @@ export abstract class BatchSubmitter {
 
   public static async getReceiptWithResubmission(
     response: TransactionResponse,
-    receiptPromises: Array<Promise<TransactionReceipt>>,
     signer: Signer,
-    numConfirmations: number,
-    resubmissionTimeout: number,
+    resubmissionConfig: ResubmissionConfig,
     log: Logger,
   ): Promise<TransactionReceipt> {
+    const {
+      numConfirmations,
+      resubmissionTimeout,
+      minGasPrice,
+      maxGasPrice,
+      gasRetryIncrement
+    } = resubmissionConfig
     const receiptPromise = response.wait(numConfirmations)
-    receiptPromises.push(receiptPromise)
-    // Wait for the tx & if it takes too long resubmit
-    const sleepAndReturnResubmit = async (timeout: number) => {
-      await new Promise((r) => setTimeout(r, timeout))
-      return 'resubmit'
+    const val = await bPromise.resolve(receiptPromise)
```

Comment:
If the initial transaction that is sent has a gas price that is too low, will this hang? I ask because it looks like we use `ynatm` after we've already submitted an initial transaction & waited for its transaction response which I imagine might not ever come

Reply:
i'll try writing a test case for this! in our original code it waits for one of the promises to resolve, so i see how this one may never return anything hmm

Reply:
Nice! I think the logic in the original should be that it waits for _either_ a promise to resolve with the receipt or a promise to resolve based on a timeout. So if the receipt takes too long the promise with the timeout will resolve and the program will continue. Hopefully that makes sense.

`ynatm` should be able to handle that logic though so this might just require not awaiting the receipt at all using the default ethers `.wait()` 

But anyway I'm a fan of adding a test case for it 😊

Reply:
@karlfloersch this ended up needing a bigger refactor to avoid making contract calls until inside `_submitAndLogTx`. lmk if this current pattern is ok! 

writing the resubmission test will need some hardhard setting tweaking, so i'll look into that tomorrow :)

Reply:
current pattern ok